bin/propolis-server/src/lib/server.rs

```diff
@@ -196,8 +196,10 @@ impl ServiceProviders {
             );
         }
         if let Some(serial_task) = self.serial_task.lock().await.take() {
-            let _ =
-                serial_task.control_ch.send(SerialTaskControlMessage::Stopping);
+            let _ = serial_task
+                .control_ch
+                .send(SerialTaskControlMessage::Stopping)
+                .await;
```

Comment:
Indeed!  I checked in with @lifning about it, who explained how propolis-cli would have handled it anyways (tolerating the "unceremonious" TCP close), as well as the circumstances which led to the bug.

Reply:
(for the curious: the `let _ =` here was originally written when `control_ch` was a `oneshot::Sender` called `close_ch`, whose `.send()` is *not* async. it got switched to an `mpsc::Sender` to support sending events besides close, and `mpsc::Sender::send` *is* async, but because of how the unused Result was suppressed here, the unused future went unreported by the compiler)